src/main/scala/swiss/dasch/domain/Asset.scala

```diff
@@ -16,7 +16,10 @@ object AssetId {
 
   def makeFromPath(file: Path): Option[AssetId] = {
     val filename = file.filename.toString
-    AssetId.make(filename.substring(0, filename.lastIndexOf("."))).toOption
+    filename.contains(".") match {
+      case true  => AssetId.make(filename.substring(0, filename.indexOf("."))).toOption
+      case false => None
+    }
```

Comment:
Could probably be rewritten to a one-liner with `Option.when(condition)(someValue)`

Reply:
not sure if 
```scala
    val filename = file.filename.toString
    Option
      .when(filename.contains("."))(filename.substring(0, filename.indexOf(".")))
      .flatMap(make(_).toOption)
```
is more readable?

The match could also be simplified into an `if ... else` like so:
```scala
    val filename = file.filename.toString
    if (filename.contains(".")) { AssetId.make(filename.substring(0, filename.indexOf("."))).toOption }
    else { None }
 ```
 
 TBH, I still prefer the current match.

Reply:
My thought was more

```scala
Option.when(filename.contains("."))(AssetId.make(filename.substring(0, filename.indexOf("."))).toOption).flatten
```

but you're right, it's probably not more readable than the current code.